### PR TITLE
add Snapshot#serialize method

### LINKED ISSUE #3144: Implement snapshot.serialize()

### PR DESCRIPTION
**Pull request by yratanov**
* fixes #3144

### CODE REVIEW

File: packages/ember-data/lib/system/snapshot.js [hunk 1]
```diff
@@ -392,6 +392,16 @@ Snapshot.prototype = {
   },
 
   /**
+    @method serialize
+    @param {Object} options
+    @return {Object} an object whose values are primitive JSON values only
+   */
+  serialize: function(options) {
+    return this.record.store.serializerFor(this.modelName).
```
Comment (igorT) [inline packages/ember-data/lib/system/snapshot.js]: you can just say this.record.serialize() and pass options

Comment (yratanov) [inline packages/ember-data/lib/system/snapshot.js]: I thought snapshot.serialize should serialize data from snapshot, not from underlying record. See my test below it won't pass if I do as you wrote.

Comment (igorT) [inline packages/ember-data/lib/system/snapshot.js]: good point.
Can you then make store.serialize use `snapshot.serialize` instead of repeating code?

Comment (yratanov) [inline packages/ember-data/lib/system/snapshot.js]: @igorT done.

**Comment by igorT** [comment]
Can you please squash, and I'll merge

File: packages/ember-data/lib/system/snapshot.js (hunk 1 above)
Comment (igorT) [inline packages/ember-data/lib/system/snapshot.js]: Can you please make this 1 line, I don't think we do this for lines this short

File: packages/ember-data/tests/integration/snapshot-test.js [hunk 2]
```diff
@@ -528,6 +528,19 @@ test("snapshot.get() proxies property to record unless identified as id, attribu
...
+test("snapshot.serialize() serializes itself", function() {
+  expect(1);
+
+  run(function() {
+    var post = env.store.push('post', { id: 1, title: 'Hello World' });
+    var snapshot = post._createSnapshot();
+
+    post.set('title', 'New Title');
+
+    deepEqual(snapshot.serialize(), { author: undefined, title: 'Hello World' }, 'shapshot serializes correctly')
```
Comment (wecc) [inline packages/ember-data/tests/integration/snapshot-test.js]: Could we add an additional assert and pass `options`?

**Comment by yratanov** [comment]
@igorT squashed 
@wecc added

PS: where can I see your style guides? except this https://github.com/emberjs/data/blob/master/CONTRIBUTING.md

**Comment by wecc** [comment]
LGTM :+1: 

@yratanov There's a style guide at https://github.com/emberjs/ember.js/blob/master/STYLEGUIDE.md

**Comment by igorT** [comment]
Thanks!

**Pull request merged**